### PR TITLE
Put files in a tarball in a subdirectory.

### PR DESCRIPTION
**Pull request by jelmer**
Fixes the first part of the issue described in #204

### CODE REVIEW

File: klaus/views.py [hunk 1]
```diff
@@ -434,8 +435,13 @@ def get_response(self):
 class DownloadView(BaseRepoView):
     """Download a repo as a tar.gz file."""
     def get_response(self):
-        tarname = "%s@%s.tar.gz" % (self.context['repo'].name,
-                                    sanitize_branch_name(self.context['rev']))
+        basename = "%s@%s" % (self.context['repo'].name,
+                              sanitize_branch_name(self.context['rev']))
+        tar_stream_arguments = {}
+        if dulwich_version >= (0, 19, 3):
```
Comment (jonashaag) [inline klaus/views.py]: Let's require Dulwich >= 0.19.3 then.

Comment (jelmer) [reply klaus/views.py]: Sure - updated.

**Comment by jonashaag** [comment]
@jelmer is 0.19.3 on PyPI already? Travis still can't find it

**Comment by jelmer** [comment]
Yep, it is now. I uploaded it to pypi just as I was pushing this branch to GitHub.

On 18 June 2018 07:29:01 BST, Jonas Haag <notifications@github.com> wrote:
>@jelmer is 0.19.3 on PyPI already? Travis still can't find it
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/jonashaag/klaus/pull/214#issuecomment-397954274

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by jonashaag** [comment]
There seems to be an issue with str/bytes

```
TypeError: Can't mix strings and bytes in path components
```

**Comment by jelmer** [comment]
Fixed; I've also updated the tarball test.

**Comment by jonashaag** [comment]
Thanks!

**Pull request merged**